package.json

```diff
@@ -26,28 +23,44 @@
     "unit-test": "jest --selectProjects=Unit"
   },
   "dependencies": {
+    "@nestjs/cli": "^9.2.0",
     "@nestjs/common": "^9.3.9",
     "@nestjs/config": "^2.3.1",
     "@nestjs/core": "^9.3.9",
     "@nestjs/passport": "^9.0.3",
     "@nestjs/platform-express": "^9.3.9",
+    "@nestjs/schematics": "^9.0.4",
     "@nestjs/swagger": "^6.2.1",
     "@nestjs/terminus": "^9.2.1",
+    "@nestjs/testing": "^9.3.9",
     "@nestjs/typeorm": "^9.0.1",
+    "@tsconfig/node18": "^1.0.1",
+    "@types/chance": "^1.1.3",
+    "@types/compression": "^1.7.2",
+    "@types/express": "^4.17.17",
+    "@types/jest": "29.4.1",
+    "@types/node": "^18.15.0",
     "class-transformer": "^0.5.1",
     "class-validator": "^0.14.0",
     "compression": "^1.7.4",
     "date-fns": "^2.29.3",
     "dotenv": "^16.0.3",
     "express-basic-auth": "^1.2.1",
+    "husky": "^8.0.3",
+    "jest": "29.5.0",
+    "lint-staged": "^13.2.0",
```

Comment:
I'm not sure we need all these packages as main packages? They are Dev dependencies

Reply:
Not having these packages in `prod` was hindering the `build` process especially when `RUN npm i --omit=dev` command is being executed as part of Dockerfile execution.

Reply:
I think something's not quite right somewhere in the build process if we need to mark these as main packages when they're dev dependencies. Maybe instead we need to:
1. Install all dependencies to node_modules
2. Build
3. Replace node_modules with just dev dependencies